writers_rollingfilewriter.go

```diff
@@ -612,8 +613,8 @@ func (rw *rollingFileWriter) Close() error {
 	return nil
 }
 
-func (rw *rollingFileWriter) tempArchiveFile() (*os.File, error) {
-	tmp := path.Join(path.Dir(rw.archivePath), ".seelog_tmp")
+func (rw *rollingFileWriter) tempArchiveFile(archiveDir string) (*os.File, error) {
+	tmp := filepath.Join(archiveDir, ".seelog_tmp")
```

Comment:
I decided to keep the ugly directory detection out of this method, because our callers should decide how to pick the directory (and they should agree on what rw.archivePath means in the future). Let me know if you disagree.